packages/components/src/tabs/index.tsx

```diff
@@ -154,26 +159,49 @@ function Tabs( {
 		setSelectedId,
 	] );
 
-	// In controlled mode, make sure browser focus follows the selected tab if
-	// the selection is changed while a tab is already being focused.
-	useLayoutEffect( () => {
-		if ( ! isControlled || ! selectOnMove ) {
+	const previousSelectedId = usePrevious( selectedId );
+
+	useEffect( () => {
+		if ( ! isControlled ) {
 			return;
 		}
 		const currentItem = items.find( ( item ) => item.id === selectedId );
 		const activeElement = currentItem?.element?.ownerDocument.activeElement;
-		const tabsHasFocus = items.some( ( item ) => {
-			return activeElement && activeElement === item.element;
-		} );
 
-		if (
+		const tabsHasFocus =
 			activeElement &&
+			items.some( ( item ) => {
+				return activeElement === item.element;
+			} );
+		const previousSelectedTabHadFocus =
+			previousSelectedId === activeElement?.id;
```

Comment:
To avoid weird edge cases in the future

```suggestion
			typeof previousSelectedId === 'string'  && previousSelectedId === activeElement?.id;
```

Reply:
Nice, I should keep this in mind too.